### PR TITLE
Test that tuples should not be part of patterns

### PR DESCRIPTION
**Pull request by jcouv**
@gafter for first review.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs [hunk 1]
```diff
@@ -4494,5 +4491,167 @@ public override ((int b, int a), (int b, int a)) M(((int a, int b), (int b, int
...
+        if (o is (1, 2)) { }
+    }
+}
+" + trivial2uple;
+
+            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.Regular.WithTuplesFeature().WithPatternsFeature());
+            comp.VerifyDiagnostics(
+                // (6,18): error CS0150: A constant value is expected
+                //         if (o is (1, 2))
+                Diagnostic(ErrorCode.ERR_ConstantExpected, "(1, 2)").WithLocation(6, 18)
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs]: This one is a little worrying. It suggests that `(1, 2)` is being parsed as a tuple expression, which should not be possible in this context.

**Comment by gafter** [comment]
LGTM

**Comment by jcouv** [comment]
@VSadov @dotnet/roslyn-compiler for a second review.

**Comment by jcouv** [comment]
@VSadov for a second review. This one is pretty simple.

**Comment by VSadov** [comment]
LGTM

**Comment by jcouv** [comment]
@dotnet-bot please test prtest/lin/dbg/unit32 again.

**Comment by jcouv** [comment]
@dotnet-bot Please test prtest/lin/dbg/unit32

**Comment by jcouv** [comment]
@dotnet-bot Please test prtest/lin/dbg/unit32

**Comment by jcouv** [comment]
@dotnet-bot Test  prtest/lin/dbg/unit32 please

**Comment by jcouv** [comment]
@dotnet-bot Test prtest/lin/dbg/unit32 please

**Pull request merged**